nomos-http-api/tests/todo.rs

```diff
@@ -0,0 +1,361 @@
+use std::{
+    net::{Ipv4Addr, SocketAddr},
+    sync::Arc,
+    time::Duration,
+};
+
+use axum::{routing, Router, Server};
+use hyper::Error;
+use nomos_http_api::{ApiService, ApiServiceSettings, Endpoint};
+use overwatch_derive::Services;
+use overwatch_rs::{overwatch::OverwatchRunner, services::handle::ServiceHandle};
+use utoipa::{
+    openapi::security::{ApiKey, ApiKeyValue, SecurityScheme},
+    Modify, OpenApi,
+};
+use utoipa_swagger_ui::SwaggerUi;
+
+use crate::todo::Store;
+
+#[derive(Services)]
+pub struct NomosApi {
+    http: ServiceHandle<ApiService<TodoEndpoint>>,
+}
+
+#[derive(OpenApi)]
+#[openapi(
+    paths(
+        todo::list_todos,
+        todo::search_todos,
+        todo::create_todo,
+        todo::mark_done,
+        todo::delete_todo,
+    ),
+    components(
+        schemas(todo::Todo, todo::TodoError)
+    ),
+    modifiers(&SecurityAddon),
+    tags(
+        (name = "todo", description = "Todo items management API")
+    )
+)]
+struct ApiDoc;
+
+struct SecurityAddon;
+
+impl Modify for SecurityAddon {
+    fn modify(&self, openapi: &mut utoipa::openapi::OpenApi) {
+        if let Some(components) = openapi.components.as_mut() {
+            components.add_security_scheme(
+                "api_key",
+                SecurityScheme::ApiKey(ApiKey::Header(ApiKeyValue::new("todo_apikey"))),
+            )
+        }
+    }
+}
+
+pub struct TodoEndpoint {
+    addr: SocketAddr,
+}
+
+#[async_trait::async_trait]
+impl Endpoint for TodoEndpoint {
```

Comment:
This would be more a `Webserver` than an endpoint right?

Reply:
In fact we could call it `Backend` so it is in the same fashion as other services (?)

Reply:
Yeah, it is a web server.